sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileAsyncApiTests.java

```diff
@@ -424,12 +433,3698 @@ public void createOptionsWithTimeToExpireRelativeToNow() {
+                null, null, true))
+            .assertNext(r -> {
+                byte[] contentMD5 = r.getHeaders().getValue(HttpHeaderName.CONTENT_MD5).getBytes();
+                try {
+                    TestUtils.assertArraysEqual(
+                        Base64.getEncoder().encode(
+                            MessageDigest.getInstance("MD5").digest(DATA.getDefaultText().substring(0, 3).getBytes())),
+                        contentMD5);
+                } catch (NoSuchAlgorithmException e) {
+                    throw new RuntimeException(e);
+                }
+            })
+            .verifyComplete();
+    }
+
+    @Test
+    public void readRetryDefault() {
+        fc.append(DATA.getDefaultBinaryData(), 0).block();
+        fc.flush(DATA.getDefaultDataSizeLong(), true).block();
+        DataLakeFileAsyncClient failureFileAsyncClient = getFileAsyncClient(getDataLakeCredential(), fc.getFileUrl(),
+            new MockFailureResponsePolicy(5));
+
+        ByteArrayOutputStream downloadData = new ByteArrayOutputStream();
+
+        StepVerifier.create(FluxUtil.collectBytesInByteBufferStream(failureFileAsyncClient.read()))
+            .assertNext(r -> {
+                try {
+                    downloadData.write(r);
+                } catch (IOException ex) {
+                    throw new UncheckedIOException(ex);
+                }
+                assertEquals(DATA.getDefaultText(), downloadData.toString());
+            })
+            .verifyComplete();
+    }
+
+    @Test
+    public void downloadFileExists() throws IOException {
+        File testFile = new File(prefix + ".txt");
+        testFile.deleteOnExit();
+        createdFiles.add(testFile);
+
+        if (!testFile.exists()) {
+            assertTrue(testFile.createNewFile());
+        }
+
+        fc.append(DATA.getDefaultBinaryData(), 0);
+        fc.flush(DATA.getDefaultDataSizeLong(), true);
+
+        // Default overwrite is false so this should fail
+        StepVerifier.create(fc.readToFile(testFile.getPath()))
+            .verifyErrorSatisfies(r -> {
+                UncheckedIOException ex = assertInstanceOf(UncheckedIOException.class, r);
+                assertInstanceOf(FileAlreadyExistsException.class, ex.getCause());
+            });
+    }
+
+    @Test
+    public void downloadFileExistsSucceeds() throws IOException {
+        File testFile = new File(prefix + ".txt");
+        testFile.deleteOnExit();
+        createdFiles.add(testFile);
+
+        if (!testFile.exists()) {
+            assertTrue(testFile.createNewFile());
+        }
+
+        fc.append(DATA.getDefaultBinaryData(), 0).block();
+        fc.flush(DATA.getDefaultDataSizeLong(), true).block();
+
+        StepVerifier.create(fc.readToFile(testFile.getPath(), true))
+            .assertNext(r -> {
+                try {
+                    assertEquals(DATA.getDefaultText(), new String(Files.readAllBytes(testFile.toPath()), StandardCharsets.UTF_8));
```

Comment:
I'd move this to outside of a StepVerifier call chain and in this assertNext just verify a response was returned and then call this after the StepVerifier call